backend/src/server.py

```diff
@@ -131,32 +125,21 @@ def update_characters_file(working_directory, request_args):
                 if line == '':
                     break;
                 info = json.loads(line);
-                info['pinyin'] = [pinyins[i]];
-                info['definition'] = definitions[i];
+                pinyin = request_body['characters'][info["character"]]['pinyin']
+                definition = request_body['characters'][info["character"]]['definition']
+                info['pinyin'] = [pinyin];
+                info['definition'] = definition;
                 f_new.write(json.dumps(info) + '\n');
                 i += 1;
     shutil.copy(new_file_path, file_path);
 
-def update_words_file(working_directory, request_args):
+def update_words_file(working_directory, request_body):
+    try:
+        words_definitions = request_body['words']
```

Comment:
If `words` parameter is not provided, no words should be generated (to match current logic). Also, this method now throws, if the number of words in words_definitions does not match the number of words in WORDS_FILE. Again, to match the current logic, we should use an empty array at words_definitions[i], for which we do not have the data in words_definitions.